qiskit/circuit/parameterexpression.py

```diff
@@ -22,6 +22,8 @@
 from qiskit.circuit.exceptions import CircuitError
 from qiskit.utils import optionals as _optionals
 
+if _optionals.HAS_SYMENGINE:
+    import symengine
```

Comment:
Actually symengine isn't that heavy to import as it's mostly cython and a statically linked c++ lib. It's sympy we normally have to avoid as it's super slow to import. The bigger thing is that symengine has a narrower platform support than qiskit and we can't rely on it being present, it's in the requirements list for platforms that support it but not all do.

Reply:
Oh, it actually appeared quite heavy on my laptop when I tried it - must have hitting a really cold cache.  At any rate, these changes were unrelated to this PR logically, so I put them back in place.